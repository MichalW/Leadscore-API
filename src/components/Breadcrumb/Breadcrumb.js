import PropTypes from 'prop-types';
import React from 'react';
import capitalize from 'lodash/capitalize';
import {Breadcrumb, Grid} from 'semantic-ui-react';
import {Link} from 'react-router-dom';

const BreadcrumbComponent = ({location: {pathname}}) => {
  const routeTitle = capitalize(pathname.slice(1));

  return (
    <Grid>
      <Grid.Row>
        <Grid.Column>
          <Breadcrumb>
            <Breadcrumb.Section link to="/" as={Link}>Leadscore</Breadcrumb.Section>

            {routeTitle && [
              <Breadcrumb.Divider icon="right angle" key="divider" />,
              <Breadcrumb.Section key="title">
                {routeTitle}
              </Breadcrumb.Section>,
            ]}
          </Breadcrumb>
        </Grid.Column>
      </Grid.Row>
    </Grid>
  );
};

BreadcrumbComponent.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }).isRequired,
};

export default BreadcrumbComponent;
