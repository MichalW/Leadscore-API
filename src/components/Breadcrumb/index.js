import {withRouter} from 'react-router-dom';

import Breadcrumb from './Breadcrumb';

export default withRouter(Breadcrumb);
