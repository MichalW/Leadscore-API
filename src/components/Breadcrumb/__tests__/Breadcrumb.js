import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';

import Breadcrumb from '../Breadcrumb';

it('should render Breadcrumb on root', () => {
  const props = {
    location: {
      pathname: '/',
    },
  };

  const contacts = shallow(<Breadcrumb {...props} />);

  expect(toJson(contacts)).toMatchSnapshot();
});


it('should render Breadcrumb on page', () => {
  const props = {
    location: {
      pathname: '/example'
    },
  };

  const contacts = shallow(<Breadcrumb {...props} />);

  expect(toJson(contacts)).toMatchSnapshot();
});
