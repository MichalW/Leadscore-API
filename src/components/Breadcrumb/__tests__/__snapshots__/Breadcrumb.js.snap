// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Breadcrumb on page 1`] = `
<Grid>
  <GridRow>
    <GridColumn>
      <Breadcrumb>
        <BreadcrumbSection
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          link={true}
          to="/"
        >
          Leadscore
        </BreadcrumbSection>
        <BreadcrumbDivider
          icon="right angle"
          key="divider"
        />
        <BreadcrumbSection
          key="title"
        >
          Example
        </BreadcrumbSection>
      </Breadcrumb>
    </GridColumn>
  </GridRow>
</Grid>
`;

exports[`should render Breadcrumb on root 1`] = `
<Grid>
  <GridRow>
    <GridColumn>
      <Breadcrumb>
        <BreadcrumbSection
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "Link",
              "propTypes": Object {
                "innerRef": [Function],
                "onClick": [Function],
                "replace": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
            }
          }
          link={true}
          to="/"
        >
          Leadscore
        </BreadcrumbSection>
      </Breadcrumb>
    </GridColumn>
  </GridRow>
</Grid>
`;
